src/rtpghi~.c

```diff
@@ -193,8 +189,16 @@ void rtpghi_tilde_setup(void) {
                               CLASS_DEFAULT,
                                A_GIMME,
                               0);
-	//to do: add methods to change do_causal_pd, tol_pd on runtime
-   class_addmethod(rtpghi_tilde_class,
-                   (t_method)rtpghi_tilde_dsp, gensym("dsp"), A_CANT, 0);
-   CLASS_MAINSIGNALIN(rtpghi_tilde_class, t_rtpghi_tilde, f);
-}
+	//to do: maybe change tol_pd on runtime (while DSP turned ON)
+	class_addbang  (rtpghi_tilde_class, rtpghi_tilde_res);
+	class_addmethod(rtpghi_tilde_class,
+			(t_method)rtpghi_tilde_causal, gensym("set_c"),
```

Comment:
i think `set_c` is really a bad name. e.g. `set_c 1` tells us, that we are **set**ing the value of **c** to **123**. but what is `c` in the first place? a *c*ounter?

otoh, having a message `causal 1` tells us to do *something* with **causal**ity and the value **1**. in the absence of any explicit specifier what this *something* is, i think it is a safe bet to assume that we are *setting* the *causal* value to *1* (which is exactly what is happening).

feel free to use cryptic variables in your implementation (i *might* still object :-)), but make sure that your public API is verbose and self-explanatory.